### DISCUSSION TITLE
Sort file names after using groupTuple

### DISCUSSION THREAD

**Discussion by zillurbmb51** [nextflow-io/nextflow#2207] [Q&A]
Hi, I have a similar issue https://github.com/nextflow-io/nextflow/issues/347 only have numbers in the file name.
The solution there:
`my_channel.groupTuple().map { chrom,bgen -> tuple( chrom, bgen.sort{it.name})}`
does not resolve the problem, because in my file name there are numbers existed. example: 
`[chr1, [chr1.step10.imputed.chunk0.bgen, chr1.step10.imputed.chunk1.bgen, chr1.step10.imputed.chunk2.bgen, chr1.step10.imputed.chunk3.bgen,.....]]`
After using 
`.map { chrom,bgen -> tuple( chrom, bgen.sort{it.name})}` it is giving me 
`[chr1, [chr1.step10.imputed.chunk0.bgen, chr1.step10.imputed.chunk1.bgen, chr1.step10.imputed.chunk10.bgen, chr1.step10.imputed.chunk11.bgen,.....]]` . As I understand, sorting order was `0,1,10,11,12,...` but I need, `0,1,2,3,4...` That means 

`[chr1, [chr1.step10.imputed.chunk0.bgen, chr1.step10.imputed.chunk1.bgen, chr1.step10.imputed.chunk2.bgen, chr1.step10.imputed.chunk3.bgen,.....]]` . How could I get that after using `groupTuple` ?

**Comment by abhi18av** [comment]
Hi @zillurbmb51 , for questions related to usage of Nextflow, please make use of the [discussions forum](https://github.com/nextflow-io/nextflow/discussions). The issues sections are used for bugs/feature request regarding Nextflow runtime. This is a new functionality of Github and similar to issues, but dedicated to only questions.

**Comment by abhi18av** [comment]
Regarding the solution, I think that we could rely on some Groovy magic and closure over here

Given the following files in the filesystem

```
chr1.step10.imputed.chunk0.bgen
chr1.step10.imputed.chunk1.bgen
chr1.step10.imputed.chunk10.bgen
chr1.step10.imputed.chunk100.bgen
chr1.step10.imputed.chunk11.bgen
chr1.step10.imputed.chunk78.bgen
```

We want to soft them based on the `NUM` in  `...chunkNUM.bgen` pattern. For this we separate the `NUM` part and apply sort based on that.

```nextfow
Channel.fromPath("*.bgen")
       .set { fasta_ch }


fasta_ch
    .toSortedList { it -> (it.name =~ /chr1.step10.imputed.chunk(.*?).bgen/)[0][1].toInteger() }
    .flatten()
    .view()



```


This yields the desired result

```
/Users/eklavya/projects/code/_scratch/nextflow_playground/chr1.step10.imputed.chunk0.bgen
/Users/eklavya/projects/code/_scratch/nextflow_playground/chr1.step10.imputed.chunk1.bgen
/Users/eklavya/projects/code/_scratch/nextflow_playground/chr1.step10.imputed.chunk10.bgen
/Users/eklavya/projects/code/_scratch/nextflow_playground/chr1.step10.imputed.chunk11.bgen
/Users/eklavya/projects/code/_scratch/nextflow_playground/chr1.step10.imputed.chunk78.bgen
/Users/eklavya/projects/code/_scratch/nextflow_playground/chr1.step10.imputed.chunk100.bgen

```

**Reply by zillurbmb51** [reply]
Thanks a lot. After using the above solution, I am getting this:
![toSortedList_error1](https://user-images.githubusercontent.com/12833907/124723759-a4f3af80-df0b-11eb-9cce-b07847b3e7d2.png)

My files are a part of tuples: `[chr1, [chr1.step10.imputed.chunk0.bgen, chr1.step10.imputed.chunk1.bgen, chr1.step10.imputed.chunk2.bgen, chr1.step10.imputed.chunk3.bgen,.....]]`
So, maybe the `toSortedList` operator need to be modified for tuples

These tuples were generated from:
`imputed.groupTuple().transpose().map{chrom,bfiles -> tuple(chrom,bfiles[0])}.groupTuple()`
Here, `imputed` is a output channel which produces multiple files for each chromosome. The `groupTuple` disrupted the ordering of files.
Need help here.

**Reply by abhi18av** [reply]
You'd need to customize the above solution I outlined, to fit your needs, for example.

```nextflow
 my_channel.groupTuple().map { chrom,bgen -> tuple( chrom, 
bgen.toSortedList { it -> (it.name =~ /chr1.step10.imputed.chunk(.*?).bgen/)[0][1].toInteger() })
}

```
